### PR TITLE
Feat(Transfer): Allow to specify item-unit

### PR DESCRIPTION
**Pull request by y-takey**
I want to specify item-unit case by case. for example, people list, book list and more.

sorry i can't write chinese so i add document to `components/transfer/index.zh-CN.md` using google translate. Please let me know if there is any weird text.

Thanks,

---

### Result for checklist

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

  * [x] Update API docs for the component.
  * [ ] ~~Update/Add demo to demonstrate new feature.~~  I think that it is trivial so it is unnecessary.
  * [x] Update TypeScript definition for the component.
  * [x] Add unit tests for the feature.

### CODE REVIEW

**Comment by yesmeck** [comment]
`Transfer` should support `locale` prop, like `Table` does https://github.com/ant-design/ant-design/blob/master/components/table/Table.tsx#L940

**Comment by y-takey** [comment]
@yesmeck Thanks for the review :)
I've supported `locale` prop, could you review it again?
May I ignore above fail of circleci? detail:
```
antd-tools run guard
/home/ubuntu/ant-design/node_modules/antd-tools/lib/gulpfile.js:7
const { execSync } = require('child_process');
      ^

SyntaxError: Unexpected token {
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:374:25)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Module.require (module.js:354:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/home/ubuntu/ant-design/node_modules/antd-tools/lib/cli/run.js:23:3)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)
```

File: components/transfer/index.tsx [hunk 1]
```diff
@@ -321,14 +320,12 @@ export default class Transfer extends React.Component<TransferProps, any> {
...
 
-  renderTransfer = (locale: TransferLocale) => {
+  renderTransfer = (transferLocale: TransferLocale) => {
     const {
       prefixCls = 'ant-transfer',
       className,
       operations = [],
       showSearch,
-      notFoundContent,
-      searchPlaceholder,
```
Comment (afc163) [inline components/transfer/index.tsx]: That will make breaking change.

**Comment by y-takey** [comment]
@afc163 Thanks for the review :)
What shoud I do next? or Is there anything I should do?

**Comment by afc163** [comment]
image

**Comment by y-takey** [comment]
I've passed all checks! 🎉

**Comment by afc163** [comment]
image

Please keep other props still working to avoid breaking change.

**Comment by y-takey** [comment]
@afc163 Thanks for your advice, I'll fix it that like!

By the way, (I had overlooked it...) Should I include the `titles` property  in `locale` property too?

**Comment by afc163** [comment]
That is not nesscessary cause `titles` don't have built-in value.

**Comment by y-takey** [comment]
OK, I've fixed to avoid breaking change.
Could you review it again?

**Pull request merged**